test/test_cross_windowize_by_event.py

```diff
@@ -0,0 +1,103 @@
+from resonance.tests.TestProcessor import TestProcessor
+import resonance.si as si
+import resonance.db as db
+import resonance.cross.windowize_by_events
+from resonance.time import timeoption2ts
+import numpy as np
+
+
+class TestCrossWindowizeByEvent(TestProcessor):
+    def setUp(self):
+        self.channels = 1
+        self.sampling_rate = 100
+        self.window_size = 11
+        self.shift = 0
+        self.drop_late_events = True
+        self.late_time = 10
+
+    def test_empty_events(self):
+        channels = 2
+        c_si = si.Channels(channels, self.sampling_rate)
+        e_si = si.Event()
+
+        streams = [c_si, e_si]
+        src_block = db.Channels(c_si, timeoption2ts(c_si, 211), np.arange(26, 76))
+
+        w_si = si.Window(channels, self.window_size, self.sampling_rate)
+        expected_block = db.Window.make_empty(w_si)
+
+        self.check_processor(streams,
+                             [src_block],
+                             {'out_0': [expected_block]},
+                             resonance.cross.windowize_by_events,
+                             self.window_size,
+                             self.shift,
+                             self.drop_late_events,
+                             self.late_time)
+
+    def test_windowize(self):
+        c_si = si.Channels(self.channels, self.sampling_rate)
+        e_si = si.Event()
+
+        streams = [c_si, e_si]
+        src_blocks = [
+            db.Channels(c_si, timeoption2ts(c_si, 201), 1),
+            db.Channels(c_si, timeoption2ts(c_si, 201 + 99), np.arange(2, 101)),
+            db.Event(e_si, timeoption2ts(c_si, 202), "first event"),
+            db.Event(e_si, timeoption2ts(c_si, 290), "last event")]
+
+        w_si = si.Window(self.channels, self.window_size, self.sampling_rate)
+        expected = [
+            db.Window.make_empty(w_si),
+            db.Window.make_empty(w_si),
+            db.Window(w_si, timeoption2ts(c_si, 201 + self.window_size), np.arange(2, 13)),
+            db.Window(w_si, timeoption2ts(c_si, 289 + self.window_size), np.arange(90, 101)),
+        ]
+
+        self.check_processor(streams,
+                             src_blocks,
+                             {'out_0': expected},
+                             resonance.cross.windowize_by_events,
+                             self.window_size,
+                             self.shift,
+                             self.drop_late_events,
+                             self.late_time)
+
+    def test_windowize_large_data(self):
```

Comment:
windowize_large_data is not explaining the problem you had, it could be something like
test_data_bigger_than_ring_buffer